lib/mox/server.ex

```diff
@@ -6,40 +6,46 @@ defmodule Mox.Server do
   # Public API
 
   def start_link(_options) do
-    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+    GenServer.start_link(__MODULE__, %{expectations: %{}, allowances: %{}}, name: __MODULE__)
   end
 
   def add_expectation(owner_pid, key, value) do
     GenServer.call(__MODULE__, {:add_expectation, owner_pid, key, value})
   end
 
-  def fetch_fun_to_dispatch(owner_pid, key) do
-    GenServer.call(__MODULE__, {:fetch_fun_to_dispatch, owner_pid, key})
+  def fetch_fun_to_dispatch(caller_pid, key) do
+    GenServer.call(__MODULE__, {:fetch_fun_to_dispatch, caller_pid, key})
   end
 
   def pending_expectations(owner_pid, for) do
     GenServer.call(__MODULE__, {:pending_expectations, owner_pid, for})
   end
 
+  def allow(mock, owner_pid, pid) do
+    GenServer.call(__MODULE__, {:allow, mock, owner_pid, pid})
+  end
+
   # Callbacks
 
   def handle_call({:add_expectation, owner_pid, key, expectation}, _from, state) do
     state = maybe_add_and_monitor_pid(state, owner_pid)
 
     state =
-      update_in state[owner_pid], fn state ->
-        Map.update(state, key, expectation, &merge_expectation(&1, expectation))
+      update_in state[:expectations][owner_pid], fn owned_expectations ->
```

Comment:
Because we expect those keys to be there, my suggestion is to always use `state.expectations` and `state.allowances`.

Reply:
ah ok. In my projects I follow the convention of struct access being always done with dot, and map access being always done with []. But it's ok, I can change that.

Reply:
If the map has fixed keys, I always use it. It helps later in case of bugs and typos.